src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs

```diff
@@ -127,70 +127,6 @@ public IEnumerable<StateSet> GetOrCreateStateSets(Project project)
                 return null;
             }
 
-            /// <summary>
-            /// Return <see cref="StateSet"/>s that are added as the given <see cref="Project"/>'s AnalyzerReferences.
-            /// This will never create new <see cref="StateSet"/> but will return ones already created.
-            /// </summary>
-            public ImmutableArray<StateSet> CreateBuildOnlyProjectStateSet(Project project)
-            {
-                var projectStateSets = project.SupportsCompilation
-                    ? GetOrUpdateProjectStateSets(project)
-                    : ProjectAnalyzerStateSets.Default;
-                var hostStateSets = GetOrCreateHostStateSets(project, projectStateSets);
```

Comment:
aside, i'd love to know what the different state sets all mean.

Reply:
HostStateSets - state sets corresponding to analyzers coming from installed analyzer VSIXes, which are applicable for any project opened in the VS Session. The most common use case is for our built-in IDE code style analyzers that ship in Features.dll inside Roslyn VSIX installed with VS.

ProjectStateSets - state sets corresponding to analyzers coming from installed analyzer NuGet packages, which are applicable only to the specific project.